homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -55,63 +64,112 @@
 )
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+async def async_setup_platform(
+    homeassistant, config, async_add_entities, discovery_info=None
+):
     """Set up the Enphase Envoy sensor."""
     ip_address = config[CONF_IP_ADDRESS]
     monitored_conditions = config[CONF_MONITORED_CONDITIONS]
     name = config[CONF_NAME]
     username = config[CONF_USERNAME]
     password = config[CONF_PASSWORD]
 
-    envoy_reader = EnvoyReader(ip_address, username, password)
+    if "inverters" in monitored_conditions:
+        envoy_reader = EnvoyReader(ip_address, username, password, inverters=True)
+    else:
+        envoy_reader = EnvoyReader(ip_address, username, password)
+
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        data = {}
+        async with async_timeout.timeout(30):
+            try:
+                await envoy_reader.getData()
+            except httpx.HTTPStatusError as err:
+                _LOGGER.error("Error retrieving data: %s", err)
+            except httpx.HTTPError as err:
+                raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+            for condition in monitored_conditions:
+                if condition != "inverters":
+                    data[condition] = await getattr(envoy_reader, condition)()
+                else:
+                    try:
+                        data["inverters_production"] = await getattr(
+                            envoy_reader, "inverters_production"
+                        )()
+                    except httpx.HTTPStatusError as err:
```

Comment:
Can we remove this error handling if there's no I/O left in this library call?

Reply:
Yeah, removed this error handling it's not needed